fuse/data/utils/collates.py

```diff
@@ -247,3 +255,22 @@ def crop_padding(
         cropped_sequences = [ids[:min_length] for ids in input_ids_list]
         batched_sequences = torch.stack(cropped_sequences, dim=0)
         return batched_sequences
+
+    @staticmethod
+    def match_length_to_target_key(
```

Comment:
Cropping?
What happens when the target_key is shorter then keys_to_match?

Reply:
this will raise an error, maybe I'll change the name length to `crop_length_to_target_key`? The motivation is when the encoder and labels should have the same length and pad cropping was applied to the encoder input - so we want to match the labels.

Reply:
I assume you mean decoder input?
I asked Michal to solve it in a different way.
(labels and decoder should always have the same length).

Reply:
sorry, I do mean encoder, this is for the case of an encoder-only model